src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -34,21 +35,31 @@ internal class AttributeCloner<TAttribute>
         // Compute the values to apply to Settable properties on newly created attribute.
         private readonly Action<TAttribute, BindingData>[] _propertySetters;
 
-        // Optional hook for post-processing the attribute. This is intended for legacy hack rules.
-        private readonly Func<TAttribute, Task<TAttribute>> _hook;
-
         private readonly Dictionary<PropertyInfo, AutoResolveAttribute> _autoResolves = new Dictionary<PropertyInfo, AutoResolveAttribute>();
 
         private static readonly BindingFlags Flags = BindingFlags.Instance | BindingFlags.Public;
 
+        private readonly AttributeClonerContext _context;
+
         public AttributeCloner(
             TAttribute source,
             BindingDataContract bindingDataContract,
-            INameResolver nameResolver = null,
-            Func<TAttribute, Task<TAttribute>> hook = null)
+            INameResolver nameResolver) : this(source, bindingDataContract, AttributeClonerContext.New(nameResolver))
         {
-            _hook = hook;
-            nameResolver = nameResolver ?? new EmptyNameResolver();
+        }
+
+        public AttributeCloner(
+            TAttribute source,
+            BindingDataContract bindingDataContract,
+            AttributeClonerContext context = null)
+        {
+            if (context == null)
+            {
+                context = new AttributeClonerContext();
```

Comment:
what is that?

---
In reply to: [114161428](https://github.com/Azure/azure-webjobs-sdk/pull/1124#discussion_r114161428) [](ancestors = 114161428)